src/basic/syslog-util.c

```diff
@@ -21,6 +21,8 @@ int syslog_parse_priority(const char **p, int *priority, bool with_facility) {
         assert(*p);
         assert(priority);
 
+        *p += strspn(*p, WHITESPACE);
```

Comment:
this change is not OK I figure. The thing is, these syslog prio prefixes are used all over the place including by the kernel kmsg logic, and there, if you get a message prefixed with `<X>` then it means: priority shall be X. But if it's not prefixed with `<X>` then it means: use default priority. Now, the in the kernel context hence a log message of "<4>foo" and " <4>foo" is very different. The leading whitepace means the latter is printed as literal " <4>foo", while the former would be printed as "foo" with 4 used as priority.

Now, the function you are changing here is used in multiple contexts. In some leading whitespace matters (as described above) in others it does not. Given that it is like it is, I figure stripping leading whitespace should really be left to the caller, and not done internally here.

Reply:
Do we actually implement following?

> But if it's not prefixed with <X> then it means: use default priority

This is what we do now right after ```assert``` statements,

```C
        if ((*p)[0] != '<')
                return 0;
```
AFAICT, return 0 means that we actually failed to parse the value and we didn't assign any priority to ```*priority```. But maybe I am missing something.

Secondly, kernel docs says that ```/dev/kmsg``` lines starting with ```' '``` are continuations. I'd assume they are not really valid w/o prior context. Even though I doubt that kernel actually checks any of that and would refuse such writes from userspace.

> A line starting with ' ', is a continuation line, adding
> key/value pairs to the log message, which provide the machine
> readable context of the message, for reliable processing in
> userspace.

Reply:
> Do we actually implement following?
> 
> >     But if it's not prefixed with then it means: use default priority

Yes, see stdout_stream_log(), which uses it in the stream logging (i.e. stdout/stderr) logging, if SyslogLevelPrefix=yes is set for a service.